### DISCUSSION TITLE
How do I run a baremetal hello world program?

### DISCUSSION THREAD

**Discussion by Jeban2** [eugene-tarassov/vivado-risc-v#68] [Q&A] [answered]
I've managed to follow the repository and run a hello world code on the Debian Linux. Now how would i go about configuring the bootloader such that it doesn't launch the Debian Linux but rather a simple bare-metal hello world program?

**Comment by eugene-tarassov** [comment, marked as answer]
You don't need to modify bootloader.

I have added simple bare metal hello world example to the repo: [bare-metal](https://github.com/eugene-tarassov/vivado-risc-v/tree/master/bare-metal). You can put in on SD card instead of Linux.

**Reply by Jeban2** [reply]
Sorry sir could you elaborate on this? `/mk-sd-card `partitions the SD card according to the debian-riscv64 directory. When i manually put the boot.elf file on the SD the console returns "cannot access SD card". Is there a specific way to do it?

**Reply by mwrnd** [reply]
`mk-sd-card` expects `vivado-risc-v/debian-riscv64/debian-riscv64.sd.img` which is generated using [`mk-sd-image`](https://github.com/eugene-tarassov/vivado-risc-v/blob/36b5433909945d1a34a0e367f55164ae1335f6f9/mk-sd-card#L6). Try overwriting `workspace/boot.elf` with a custom version. [`mk-sd-image`](https://github.com/eugene-tarassov/vivado-risc-v/blob/36b5433909945d1a34a0e367f55164ae1335f6f9/mk-sd-image) makes use of `workspace/boot.elf` in two places but it is easier to just try overwriting it first.

**Comment by DuyTrandeLion** [comment]
@eugene-tarassov What if my board does not have an SD card? I ported the project to the Arty-A7-100 board but I don't know how to test it.

**Reply by mwrnd** [reply]
Refer to [this thread](https://github.com/eugene-tarassov/vivado-risc-v/discussions/104#discussioncomment-3866234) for help with testing the `bare-metal` demo.

**Comment by davekeeshan** [comment]
I have bare metal working on riscv64 on an arty7-100, what is required, or what needs to be changed to get it working on riscv32?

I have compiled the riscv-toolchain from source, and this riscv64 is enough to get a working "Hello world", I also have access to riscv32 through this path.

**Comment by eugene-tarassov** [comment]
The README in [bare-metal](https://github.com/eugene-tarassov/vivado-risc-v/tree/master/bare-metal) describes both 64-bit and 32-bit cases.

**Reply by davekeeshan** [reply]
So it does, my mistake, thanks for that.
